sootup.analysis/src/test/java/sootup/analysis/interprocedural/ifds/ICFGCallGraphTest.java

```diff
@@ -0,0 +1,112 @@
+package sootup.analysis.interprocedural.ifds;
+
+
+import org.junit.Test;
+import sootup.analysis.interprocedural.icfg.JimpleBasedInterproceduralCFG;
+import sootup.core.model.SootClass;
+import sootup.java.bytecode.inputlocation.JavaClassPathAnalysisInputLocation;
+import sootup.java.core.JavaIdentifierFactory;
+import sootup.java.core.JavaProject;
+import sootup.java.core.language.JavaLanguage;
+import sootup.java.core.types.JavaClassType;
+
+public class ICFGCallGraphTest extends IFDSTaintTestSetUp {
+
+  @Test
+  public void ICFGDotExportTest() {
+    JavaProject javaProject =
+        JavaProject.builder(new JavaLanguage(8))
+            .addInputLocation(
+                new JavaClassPathAnalysisInputLocation(
+                    System.getProperty("java.home") + "/lib/rt.jar"))
+            .addInputLocation(
+                new JavaClassPathAnalysisInputLocation("src/test/resources/taint/binary"))
+            .build();
+
+    view = javaProject.createView();
+
+    JavaIdentifierFactory identifierFactory = JavaIdentifierFactory.getInstance();
+    JavaClassType mainClassSignature = identifierFactory.getClassType("ICFGExample");
+
+    SootClass<?> sc = view.getClass(mainClassSignature).get();
+    entryMethod =
+        sc.getMethods().stream().filter(e -> e.getName().equals("entryPoint")).findFirst().get();
+
+    entryMethodSignature = entryMethod.getSignature();
+
+    JimpleBasedInterproceduralCFG icfg =
+        new JimpleBasedInterproceduralCFG(view, entryMethodSignature, false, false);
+    String expectedCallGraph =
+        "digraph G {\n"
```

Comment:
Maybe instead of comparing the full result. you could check if all methods blocks are contained in the block and that they are connected by the calls.
Maybe you remove found parts from the string till the string is empty or you cant find a edge or code block in it.
this way it might be robust for small changes in order or something